client/my-sites/posts/post-list.jsx

```diff
@@ -301,4 +301,9 @@ var Posts = React.createClass( {
 	}
 } );
 
-module.exports = PostList;
+export default connect(
+	( state ) => ( {
+		selectedSiteId: getSelectedSiteId( state ),
+		hasSites: !! getSites( state ).length
```

Comment:
This will calculate all sites calling `getSite` for all of them, which is unnecessary for just counting the sites. Maybe we should add a separate selector for this one? Should be a pretty straightforward one.